src/arrayMethodReduce.js

```diff
@@ -5,8 +5,23 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let prev;
+    let fromIndex;
+
+    if (arguments.length < 2) {
+      prev = this[0];
+      fromIndex = 1;
+    } else {
+      prev = initialValue;
+      fromIndex = 0;
+    }
```

Comment:
```suggestion
    let prev = initialValue;
    let fromIndex = 0;

    if (arguments.length < 2) {
      prev = this[0];
      fromIndex = 1;
    } 
```

You can also set these values by default and remove the else block to make code more simpler